internal/api/client.go

```diff
@@ -1,17 +1,47 @@
 package api
 
+import (
+	"net/http"
+)
+
+// client is a client for the test splitter API.
+// It contains the organization slug, server base URL, and an HTTP client.
 type client struct {
-	ServerBaseUrl string
-	AccessToken   string
+	OrganizationSlug string
+	ServerBaseUrl    string
+	httpClient       *http.Client
 }
 
+// ClientConfig is the configuration for the test splitter API client.
 type ClientConfig struct {
-	ServerBaseUrl string
-	AccessToken   string
+	AccessToken      string
+	OrganizationSlug string
+	ServerBaseUrl    string
+}
+
+// authTransport is a middleware for the HTTP client.
+type authTransport struct {
+	accessToken string
 }
 
+// RoundTrip adds the Authorization header to all requests made by the HTTP client.
+func (t *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Wow How this worked? I didn't see you explicitly call this method. I looked at your testing file and it seems when you call `c.httpClient.Get(svr.URL)` the auth header will just be added.

Reply:
We set `authTransport` as the `Transport` for `http.Client` when we create it within `api.NewClient` function. 
Every time we make a request using that http client, it will call the `RoundTrip` function.